supporting other activities

I would like to know what entails supporting other activities (most of which done have gps data (swim, pilates), and some do (mountain bike)). 

I created a small script to see how many of my activites would be converted:

```
#!/bin/bash
MYTMP=$(mktemp)
# echo "Temp file: $MYTMP"

for FILEJSON in ../Sport-sessions/*.json; do
    type=`cat $FILEJSON | jq -c '.sport_type_id'`
    echo $type >> $MYTMP
    # echo $type 
done

echo "Count and activity ids:"
sort $MYTMP | uniq -c | sort -bgr
```

outputs something like:
> Count and activity ids:
>     537 "18"
>     319 "34"
>     262 "31"
>     198 "23"
>     106 "4"
>      75 "1"
>      59 "27"
>      47 "3"
>      34 "5"
>      26 "7"
>      20 "19"
>       5 "15"

so I would like to get most of them covered. I am not proficient in js.
thanks!

no, I get weird errors:

```
C:\temp\runtastic-gpx-master>node ./cli c:\temp c:\out                                         
internal/modules/cjs/loader.js:628                                                             
    throw err;                                                                                 
    ^                                                                                          
                                                                                               
Error: Cannot find module 'meow'                                                               
Require stack:                                                                                 
- C:\temp\runtastic-gpx-master\cli.js                                                          
←[90m    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:625:15)←[39m      
←[90m    at Function.Module._load (internal/modules/cjs/loader.js:527:27)←[39m                 
←[90m    at Module.require (internal/modules/cjs/loader.js:683:19)←[39m                        
←[90m    at require (internal/modules/cjs/helpers.js:16:16)←[39m                               
    at Object.<anonymous> (C:\temp\runtastic-gpx-master\cli.js:4:14)                           
←[90m    at Module._compile (internal/modules/cjs/loader.js:777:30)←[39m                       
←[90m    at Object.Module._extensions..js (internal/modules/cjs/loader.js:788:10)←[39m         
←[90m    at Module.load (internal/modules/cjs/loader.js:643:32)←[39m                           
←[90m    at Function.Module._load (internal/modules/cjs/loader.js:556:12)←[39m                 
←[90m    at Function.Module.runMain (internal/modules/cjs/loader.js:840:10)←[39m {             
  code: ←[32m'MODULE_NOT_FOUND'←[39m,                                                          
  requireStack: [ ←[32m'C:\\temp\\runtastic-gpx-master\\cli.js'←[39m ]                         
}                                             
```                                

Anyway, I'll send the PR as is, the patch is simple enough...
thanks

--- a/README.md
+++ b/README.md
@@ -33,15 +33,63 @@ $ npx runtastic-gpx ~/Desktop/export-20190101-000 ~/Downloads/export
 
 ## Supported types
 
-- Ride
-- Road Bike
 - Run
-- Indoor Ride
+- Nordic Walking
+- Cycling
+- Mountain Biking
+- Other
+- Skating
+- Hiking
+- Cross Country Skiing
+- Skiing
+- Snow Boarding
+- Motorbiking
+- Driving
+- Snowshoeing
 - Indoor Run
+- Indoor Ride
 - Elliptical
-- Weight Training
-- Crossfit
+- Rowing
+- Swimming
 - Walk
+- Riding
+- Golfing
+- Race Cycling
+- Tennis
+- Badminton
+- Sailing
+- Windsurfing
+- Pilates
+- Climbing
+- Frisbee
+- Weight Training
+- Volleyball
+- Handbike
+- Cross Skating
+- Soccer
+- Smovey Walking
+- Nordic Cross Skating
+- Surfing
+- Kite Surfing
+- Kayaking
+- Paragliding
+- Wake Boarding
+- Freecrossen
+- Back Country Skiing
+- Ice Skating
+- Sledding
+- Snowman Building
+- Snowball Fight
+- Curling
+- Ice Stock
+- Biathlon
+- Kite Skiing
+- Speed Skiing
+- Crossfit
+- Ice Hockey
+- Skateboarding
+- Rugby
+- Standup Paddling
 
 ## Did it work well for you?
 

--- a/index.js
+++ b/index.js
@@ -295,7 +295,7 @@ const start = async ([exportPath, outputPath = `${process.cwd()}/export`]) => {
   const exportAllSessions = async data => {
     await exportSession(data, 'run');
     await exportSession(data, 'nordic_walking');
-    await exportSession(data, 'cycling');
+    await exportSession(data, 'ride');
     await exportSession(data, 'mountain_biking');
     await exportSession(data, 'other');
     await exportSession(data, 'skating');

--- a/index.js
+++ b/index.js
@@ -7,7 +7,7 @@ const FILES_PER_SEGMENT = 25;
 const SPORT_TYPES = {
   run: '1',
   nordic_walking: '2',
-  cycling: '3',
+  ride: '3',
   mountain_biking: '4',
   other: '5',
   skating: '6',
@@ -264,7 +264,7 @@ const start = async ([exportPath, outputPath = `${process.cwd()}/export`]) => {
         .filter(
           session =>
             !(
-              ((type === 'cycling' || type === 'run') &&
+              ((type === 'ride' || type === 'run') &&
                 session.gps.length === 0) ||
               ((type === 'indoor_ride' || type === 'indoor_run') &&
                 session.heart_rate.length === 0)

--- a/index.js
+++ b/index.js
@@ -37,7 +37,7 @@ const SPORT_TYPES = {
   weight_training: '34',
   volleyball: '35',
   handbike: '36',
-  cross_Skating: '37',
+  cross_skating: '37',
   soccer: '38',
   smovey_walking: '39',
   nordic_cross_skating: '41',
@@ -325,7 +325,7 @@ const start = async ([exportPath, outputPath = `${process.cwd()}/export`]) => {
     await exportSession(data, 'weight_training');
     await exportSession(data, 'volleyball');
     await exportSession(data, 'handbike');
-    await exportSession(data, 'cross_Skating');
+    await exportSession(data, 'cross_skating');
     await exportSession(data, 'soccer');
     await exportSession(data, 'smovey_walking');
     await exportSession(data, 'nordic_cross_skating');

--- a/index.js
+++ b/index.js
@@ -5,16 +5,66 @@ const util = require('util');
 
 const FILES_PER_SEGMENT = 25;
 const SPORT_TYPES = {
-  basketball: '45', // Unsupported
-  crossfit: '69',
-  elliptical: '16',
-  indoor_ride: '15',
-  indoor_run: '14',
-  ride: '3',
-  roadbike: '22',
   run: '1',
+  nordic_walking: '2',
+  cycling: '3',
+  mountain_biking: '4',
+  other: '5',
+  skating: '6',
+  hiking: '7',
+  cross_country_skiing: '8',
+  skiing: '9',
+  snow_boarding: '10',
+  motorbiking: '11',
+  driving: '12',
+  snowshoeing: '13',
+  indoor_run: '14',
+  indoor_ride: '15',
+  elliptical: '16',
+  rowing: '17',
+  swimming: '18',
   walk: '19',
+  riding: '20',
+  golfing: '21',
+  race_cycling: '22',
+  tennis: '23',
+  badminton: '24',
+  sailing: '29',
+  windsurfing: '30',
+  pilates: '31',
+  climbing: '32',
+  frisbee: '33',
   weight_training: '34',
+  volleyball: '35',
+  handbike: '36',
+  cross_Skating: '37',
+  soccer: '38',
+  smovey_walking: '39',
+  nordic_cross_skating: '41',
+  surfing: '42',
+  kite_surfing: '43',
+  kayaking: '44',
+  basketball: '45',
+  paragliding: '47',
+  wake_boarding: '48',
+  freecrossen: '49',
+  diving: '50',
+  back_country_skiing: '53',
+  ice_skating: '54',
+  sledding: '55',
+  snowman_building: '56',
+  snowball_fight: '57',
+  curling: '58',
+  ice_stock: '59',
+  biathlon: '60',
+  kite_skiing: '61',
+  speed_skiing: '62',
+  baseball: '68',
+  crossfit: '69',
+  ice_hockey: '71',
+  skateboarding: '72',
+  rugby: '75',
+  standup_paddling: '76',
 };
 
 const parse = buffer => {
@@ -214,7 +264,7 @@ const start = async ([exportPath, outputPath = `${process.cwd()}/export`]) => {
         .filter(
           session =>
             !(
-              ((type === 'ride' || type === 'run') &&
+              ((type === 'cycling' || type === 'run') &&
                 session.gps.length === 0) ||
               ((type === 'indoor_ride' || type === 'indoor_run') &&
                 session.heart_rate.length === 0)
@@ -243,15 +293,63 @@ const start = async ([exportPath, outputPath = `${process.cwd()}/export`]) => {
   };
 
   const exportAllSessions = async data => {
-    await exportSession(data, 'ride');
     await exportSession(data, 'run');
-    await exportSession(data, 'roadbike');
-    await exportSession(data, 'indoor_ride');
+    await exportSession(data, 'nordic_walking');
+    await exportSession(data, 'cycling');
+    await exportSession(data, 'mountain_biking');
+    await exportSession(data, 'other');
+    await exportSession(data, 'skating');
+    await exportSession(data, 'hiking');
+    await exportSession(data, 'cross_country_skiing');
+    await exportSession(data, 'skiing');
+    await exportSession(data, 'snow_boarding');
+    await exportSession(data, 'motorbiking');
+    await exportSession(data, 'driving');
+    await exportSession(data, 'snowshoeing');
     await exportSession(data, 'indoor_run');
+    await exportSession(data, 'indoor_ride');
     await exportSession(data, 'elliptical');
-    await exportSession(data, 'weight_training');
-    await exportSession(data, 'crossfit');
+    await exportSession(data, 'rowing');
+    await exportSession(data, 'swimming');
     await exportSession(data, 'walk');
+    await exportSession(data, 'riding');
+    await exportSession(data, 'golfing');
+    await exportSession(data, 'race_cycling');
+    await exportSession(data, 'tennis');
+    await exportSession(data, 'badminton');
+    await exportSession(data, 'sailing');
+    await exportSession(data, 'windsurfing');
+    await exportSession(data, 'pilates');
+    await exportSession(data, 'climbing');
+    await exportSession(data, 'frisbee');
+    await exportSession(data, 'weight_training');
+    await exportSession(data, 'volleyball');
+    await exportSession(data, 'handbike');
+    await exportSession(data, 'cross_Skating');
+    await exportSession(data, 'soccer');
+    await exportSession(data, 'smovey_walking');
+    await exportSession(data, 'nordic_cross_skating');
+    await exportSession(data, 'surfing');
+    await exportSession(data, 'kite_surfing');
+    await exportSession(data, 'kayaking');
+    await exportSession(data, 'paragliding');
+    await exportSession(data, 'wake_boarding');
+    await exportSession(data, 'freecrossen');
+    await exportSession(data, 'back_country_skiing');
+    await exportSession(data, 'ice_skating');
+    await exportSession(data, 'sledding');
+    await exportSession(data, 'snowman_building');
+    await exportSession(data, 'snowball_fight');
+    await exportSession(data, 'curling');
+    await exportSession(data, 'ice_stock');
+    await exportSession(data, 'biathlon');
+    await exportSession(data, 'kite_skiing');
+    await exportSession(data, 'speed_skiing');
+    await exportSession(data, 'crossfit');
+    await exportSession(data, 'ice_hockey');
+    await exportSession(data, 'skateboarding');
+    await exportSession(data, 'rugby');
+    await exportSession(data, 'standup_paddling');
 
     console.log(
       `\n${chalk.green(

